src/pep257.py

```diff
@@ -679,73 +678,496 @@ def __getattr__(self, item):
+
+    @staticmethod
+    def _create_run_config(options):
+        """Create a `RunConfiguration` object from `options`."""
+        values = dict([(opt, getattr(options, opt)) for opt in
+                       RunConfiguration._fields])
+        return RunConfiguration(**values)
+
+    @classmethod
+    def _create_check_config(cls, options, use_dafaults=True):
+        """Create a `CheckConfiguration` object from `options`.
+
+        If `use_dafaults`, any of the match options that are `None` will
+        be replaced with their default value and the default convention will be
+        set for the checked codes.
+
+        """
+        match = cls.DEFAULT_MATCH_RE \
+            if options.match is None and use_dafaults \
+            else options.match
+
+        match_dir = cls.DEFAULT_MATCH_DIR_RE \
+            if options.match_dir is None and use_dafaults \
+            else options.match_dir
+
+        checked_codes = None
+
+        if cls._has_exclusive_option(options) or use_dafaults:
+            checked_codes = cls._get_checked_errors(options)
+
+        return CheckConfiguration(checked_codes=checked_codes,
+                                  match=match, match_dir=match_dir)
+
+    @classmethod
+    def _get_config_file_in_folder(cls, path):
+        """Look for a configuration file in `path`.
+
+        If exists return it's full path, otherwise None.
+
+        """
+        if os.path.isfile(path):
+            path = os.path.dirname(path)
+
+        for fn in cls.PROJECT_CONFIG_FILES:
+            config = RawConfigParser()
+            full_path = os.path.join(path, fn)
+            if config.read(full_path) and config.has_section('pep257'):
+                return full_path
+
+    @staticmethod
+    def _get_exclusive_error_codes(options):
+        """Extract the error codes from the selected exclusive option."""
+        codes = set(ErrorRegistry.get_error_codes())
+        checked_codes = None
+
+        if options.ignore is not None:
+            checked_codes = codes - options.ignore
+        elif options.select is not None:
+            checked_codes = options.select
+        elif options.convention is not None:
+            checked_codes = getattr(conventions, options.convention)
+
+        # To not override the conventions nor the options - copy them.
+        return copy.deepcopy(checked_codes)
+
+    @staticmethod
+    def _set_add_options(checked_codes, options):
+        """Set `checked_codes` by the `add_ignore` or `add_select` options."""
+        checked_codes |= options.add_select
+        checked_codes -= options.add_ignore
+
+    @classmethod
+    def _get_checked_errors(cls, options):
+        """Extract the codes needed to be checked from `options`."""
+        checked_codes = cls._get_exclusive_error_codes(options)
+        if checked_codes is None:
+            checked_codes = cls.DEFAULT_CONVENTION
+
+        cls._set_add_options(checked_codes, options)
+
+        return checked_codes - set('')
```

Comment:
If a configuration was an empty list, `''.split(',')` will return `['']` - this clears this.
I moved to a more reasonable location.